lib/modules/manager/git-submodules/extract.spec.ts

```diff
@@ -1,69 +1,89 @@
 import is from '@sindresorhus/is';
 import { mock } from 'jest-mock-extended';
-import _simpleGit, {
-  Response,
-  SimpleGit,
-  SimpleGitFactory,
-  TaskOptions,
-} from 'simple-git';
+import { Response, SimpleGit, SimpleGitFactory, simpleGit } from 'simple-git';
 import { GlobalConfig } from '../../../config/global';
 import * as hostRules from '../../../util/host-rules';
-import type { PackageFileContent } from '../types';
 import { extractPackageFile } from '.';
 
 jest.mock('simple-git');
-const simpleGit: jest.Mock<Partial<SimpleGit>> = _simpleGit as never;
+const simpleGitFactoryMock = simpleGit as jest.Mock<Partial<SimpleGit>>;
 // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
 const Git = jest.requireActual('simple-git') as SimpleGitFactory;
 
 describe('modules/manager/git-submodules/extract', () => {
   // flaky ci tests
   //jest.setTimeout(10 * 1000);
+  let gitMock: jest.MockedObject<
+    Pick<SimpleGit, 'env' | 'listRemote' | 'subModule' | 'raw'>
+  >;
```

Comment:
why not using partial here too? we also have a `partial<T>(...)` helper function wich accepts a partial type and fakes the full type.

Reply:
The main problem I had with using `partial` there is that it then later complaints that functions might be undefined later and thus I would need to add an explicit check around each function, which I think is more noise than just picking the attributes from

Reply:
that's wrong, the partial helper function returns the exact T. so if the function is required inside T then it's also required on the result.

remember the partial util function is not the `Partial` typescript interface!

Reply:
I found now a way to avoid using custom type definitions at all and just relay on the inherited type from (mock) from the implementation. That is probably anyway the best solution as I don't relay then on any custom types.